src/dataplugin.ts

```diff
@@ -59,53 +59,43 @@ class DataPlugin {
 
         if (fs.existsSync(this.scriptPath)) {
             throw new FileExistsError(`${config.extPrefix}DataPlugin already exists`);
-        } else {
-            // eslint-disable-next-line @typescript-eslint/no-floating-promises
-            this.createMainPy();
         }
     }
 
-    /**
-     * @param newName New name of script file. Provide name without file extension
-     */
-    public renameDataPluginScript(newName: string): void {
-        const scriptPath = this.scriptPath;
-        const newScriptPath = path.join(path.dirname(scriptPath), `${newName}.py`);
-        fs.renameSync(scriptPath, newScriptPath);
-        this.scriptPath = newScriptPath;
-    }
-
-    /**
-     * @param substr A String that is to be replaced by newSubstr
-     * @param newSubStr Replacement string
-     */
-    public replaceStringInScript(substr: string, newSubStr: string): void {
-        const scriptPath = this.scriptPath;
-        let content = fs.readFileSync(scriptPath, { encoding: 'utf8' });
-        content = content.replace(substr, newSubStr);
-        fs.writeFileSync(scriptPath, content);
+    public static async createDataPlugin(
+        name: string,
+        baseTemplate: string,
+        language: Languages
+    ): Promise<DataPlugin> {
+        const dataPlugin = new DataPlugin(name, baseTemplate, language);
+        await DataPlugin.prepareWorkspace(dataPlugin);
+        return dataPlugin;
     }
 
-    public async createMainPy(): Promise<void> {
+    public static async prepareWorkspace(dataPlugin: DataPlugin): Promise<void> {
```

Comment:
I'm just curious, what's the rationale for this method to be static instead of a class method of `DataPlugin`?

Reply:
Right, it doesn't need to be static. thx

Reply:
Cool, I didn't know that the method can be private and you can still call it from static class methods!